src/main/java/graphql/nadel/ServiceExecutionFactory.java

```diff
@@ -0,0 +1,26 @@
+package graphql.nadel;
+
+import graphql.PublicSpi;
+import graphql.schema.idl.TypeDefinitionRegistry;
+
+@PublicSpi
+public interface ServiceExecutionFactory {
+
+    /**
+     * Called to to get a function that can be called to get data for the named service
+     *
+     * @param serviceName the name of the service
+     *
+     * @return a function that can be called to get data from that service
+     */
+    ServiceExecution getDelegatedExecution(String serviceName);
```

Comment:
should rename that too `getSeriveExecution` at one point

Reply:
done